NHLBI_BioData_Catalyst/python/5_LongitudinalData.ipynb

```diff
@@ -34,476 +34,530 @@
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "4dfdf3fa",
+   "id": "a58b1890",
    "metadata": {},
    "source": [
-    "First, the longitudinal concept paths that have exam or visit number information will be saved to `longitudinal_concept_paths`."
+    "First, we will filter our dataframe containing variables which are related to `lipid` or `triglyceride` in Framingham Heart Study to those which have the keywords `exam #` or `visit #`."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "afbab004",
+   "id": "7ef1e0c6",
    "metadata": {},
    "outputs": [],
    "source": [
-    "longitudinal_concept_paths = [] # Initialize empty list\n",
-    "for i in lipid_vars['name']:\n",
-    "    # If variable has exam # or visit #, add it to longitudinal_concept_paths\n",
-    "    if re.search('(exam \\d+|visit \\d+)', i, re.IGNORECASE):\n",
-    "        longitudinal_concept_paths.append(i)\n",
-    "len(longitudinal_concept_paths)"
+    "filtered_lipid_dataframe = filtered_lipid_dataframe[filtered_lipid_dataframe.description.str.contains('exam|visit', case = False)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5f714e7f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['description'].str.extract('(exam \\d+|visit \\d+)', flags = re.IGNORECASE)[0]"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "7a777f86",
+   "id": "d0b2529f",
    "metadata": {},
    "source": [
-    "Now `longitudinal_concept_paths` will be used to extract the longitudinal variables from the `lipid_vars` dataframe. The exam or visit number will be extracted into the `exam_number` column and the longitudinal variable will be saved to the `longvar` column."
+    "Next, we will extract the exam or visit number of each variable into column `exam_number`."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "1c04d958",
+   "id": "c145293b",
    "metadata": {
-    "scrolled": true
+    "scrolled": true,
+    "tags": []
    },
    "outputs": [],
    "source": [
-    "# Save variables that have longitudinal_concept_path in it\n",
-    "long_info = lipid_vars[lipid_vars['name'].isin(longitudinal_concept_paths)].reset_index()\n",
     "# Save exam # as exam_number\n",
-    "long_info['exam_number'] = long_info['name'].str.extract('(exam \\d+|visit \\d+)', flags=re.IGNORECASE)\n",
-    "# Save variable name without exam # as longvar\n",
-    "long_info['longvar'] = long_info['name'].str.replace('(exam \\d+|visit \\d+)', 'exam', flags=re.IGNORECASE).str.lower()\n",
-    "long_info"
+    "filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['description'].str.extract('(exam \\d+|visit \\d+)', flags = re.IGNORECASE)\n",
```

Comment:
This cell also works, but gives a warning as well:

/tmp/ipykernel_1167/1612388120.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['description'].str.extract('(exam \d+|visit \d+)', flags = re.IGNORECASE)
/tmp/ipykernel_1167/1612388120.py:3: FutureWarning: The default value of regex will change from True to False in a future version.
  filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['exam_number'].str.replace('(exam|visit)', '', flags = re.IGNORECASE).str.lower()
/tmp/ipykernel_1167/1612388120.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['exam_number'].str.replace('(exam|visit)', '', flags = re.IGNORECASE).str.lower()
/tmp/ipykernel_1167/1612388120.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['exam_number'].astype('int')